src/abstractdataframe/abstractdataframe.jl

```diff
@@ -685,76 +685,49 @@ Base.vcat(df::AbstractDataFrame) = df
 Base.vcat(dfs::AbstractDataFrame...) = vcat(AbstractDataFrame[dfs...])
 
 Base.vcat(dfs::Vector{Void}) = dfs
+
+_isnullable{A<:AbstractArray}(::Type{A}) = eltype(A) <: Nullable
+
 function Base.vcat{T<:AbstractDataFrame}(dfs::Vector{T})
     isempty(dfs) && return DataFrame()
-    coltyps, colnams, similars = _colinfo(dfs)
-
     res = DataFrame()
-    Nrow = sum(nrow, dfs)
-    for j in 1:length(colnams)
-        colnam = colnams[j]
-        col = similar(similars[j], coltyps[j], Nrow)
-
-        i = 1
-        for df in dfs
-            if haskey(df, colnam)
-                copy!(col, i, df[colnam])
-            end
-            i += size(df, 1)
+    nrows = sum(nrow, dfs)
+    for colnam in unique(Base.flatten(names.(dfs)))
```

Comment:
`flatten` lives under `Base.Iterators` on Julia 0.6. You can just test this using `isdefined` and do `using Base: flatten` or `using Base.Iterators: flatten` at the top of the file.